### DISCUSSION TITLE
reverse_proxy function causes WARNING:asyncio:socket.send() raised exception when proxying video urls

### DISCUSSION THREAD

**Discussion by gjmhmm8** [fastapi/fastapi#10721] [Questions]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.head("/proxy={url_path:path}", dependencies=[Depends(login_check)])
@app.get("/proxy={url_path:path}", dependencies=[Depends(login_check)])
async def reverse_proxy(url_path: str):
    # Adapted from: https://github.com/tiangolo/fastapi/issues/1788
    url = httpx.URL(url_path)
    headers = {}
    if Context.hf_token is not None:
        headers["Authorization"] = f"Bearer {Context.hf_token}"
    rp_req = client.build_request("GET", url, headers=headers)
    rp_resp = await client.send(rp_req, stream=True)
    return StreamingResponse(
        rp_resp.aiter_raw(),
        status_code=rp_resp.status_code,
        headers=rp_resp.headers,  # type: ignore
        background=BackgroundTask(rp_resp.aclose),
    )
```

### Description

I am using the reverse_proxy function from [this comment] to proxy video urls in my fastapi app. 
However, when I try to use it to proxy a video url,  I get a lot of warnings like this:
WARNING:asyncio:socket.send() raised exception.

### Operating System

Windows

### FastAPI Version

0.104.1

### Pydantic Version

1.9.1

### Python Version

Python 3.9.0

### Additional Context

httpx 0.25.2
use bilibili video url

**Comment by gjmhmm8** [comment]
@tiangolo

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
